ethbloom/Cargo.toml

```diff
@@ -12,12 +12,15 @@ repository = "https://github.com/debris/ethbloom"
 tiny-keccak = "1.4"
 crunchy = { version = "0.1.6", features = ["limit_256"] }
 fixed-hash = { version = "0.1.1", path = "../fixed-hash" }
+ethereum-types-serialize = { version = "0.1", path = "../serialize", optional = true }
+serde = { version = "1.0", optional = true }
 
 [dev-dependencies]
 rand = "0.4"
 rustc-hex = "1.0"
 
 [features]
-default = ["std", "heapsizeof"]
+default = ["std", "heapsizeof", "serialize"]
```

Comment:
Should this be a default feature? Why not just `serde`?
Similar pattern:
https://github.com/rust-num/num/blob/master/Cargo.toml#L48

Reply:
because ethbloom depends on a crate `ethereum-types-serialize` which contains utility functions **and** imports serde. So there cannot be a feature with the same name a optional dependency.

Reply:
Are you sure it won't work? Have a look at the linked `rust-num` crate `Cargo.toml` file.
IMHO:
```
serde = [
  "std",
  "ethereum-types-serialize"
]
```
should work fine

Reply:
but num is not using `serde` crate itself. we are. in our case cargo would fail with:

```
error: failed to parse manifest at `/Users/marek/projects/ethcore/primitives/ethereum-types/Cargo.toml`

Caused by:
  Features and dependencies cannot have the same name: `serde`
```